exonum/src/api/mod.rs

```diff
@@ -237,6 +250,22 @@ impl ServiceApiBuilder {
     pub fn private_scope(&mut self) -> &mut ServiceApiScope {
         &mut self.private_scope
     }
+
+    /// Returns an optional reference to the Blockchain.
+    pub fn blockchain(&self) -> Option<&Blockchain> {
+        self.blockchain.as_ref()
+    }
+
+    /// Returns an optional reference to the ApiSender.
+    pub fn api_sender(&self) -> Option<&ApiSender> {
+        self.blockchain().map(|blockchain| &blockchain.api_sender)
+    }
+
+    /// Returns an optional value to the PublicKey.
+    pub fn public_key(&self) -> Option<PublicKey> {
+        self.blockchain()
+            .map(|blockchain| blockchain.service_keypair.0)
+    }
```

Comment:
I'd rephrase the documentation comments to explain when `Some` values are expected from these getters. For example, "Returns service public key of the blockchain if blockchain is set" or something like that. It's obvious from return types that the getters return optional values. Developers are more interested when they can do something with these values.

Reply:
@ilammy as for me state: "if blockchain is set" isn't more clear.

Probably we should add info, that this `Option` is safe to unwrap inside `wire` method?

Reply:
@vldm I suppose it can be safe if `builder.blockchain().is_some()`